http-gateway/web/src/index.js

```diff
@@ -37,7 +37,7 @@ fetch('/web_configuration.json')
       // only set the code to the session storage, so that the caller can process it.
       const urlParams = new URLSearchParams(window.location.search)
       const code = urlParams.get('code')
-      if (window.location.pathname === '/things' && code) {
+      if (window.location.pathname === '/' && code) {
```

Comment:
This is the part which I mentioned you in the chat. Please, change the `/` to `/devices`, I've already registered this URL in the client so it will work (at least locally). To give you a bit more about where it comes from, this is the code which renders an iframe with the provided URL (which contains the /devices) https://github.com/plgd-dev/hub/pull/726/files#diff-98731371d18057ad29d7bb7d0f04a52fb2f0da9f2366e8c4ebe7c958d517f136R150 .

Notice that the URL contains a get parameter `code`, which indicated that we are requesting an authorization code. This goes through the `deviceOauthClient`, that's why we have to make sure, this URL is registered in the client as a redirect URI.

This `if` statement here makes sure, that if the URL does contain `/devices` and `code` as a get parameter (which it will, because that's how we get redirected back from the auth server), it stores the `code` value into the sessions storage, and the Promise which rendered the iframe will check for few seconds for that stored variable. This was the only way I was able to implement this, cause the API is not fetchable...

@ondrejtomcik Make sure you update the clients later.

Reply:
Will do